agent/charms/testflinger-agent-charm/src/charm.py

```diff
@@ -93,106 +91,114 @@ def _run_with_logged_errors(self, cmd):
         except FileNotFoundError:
-            logger.error("No systemd unit file found when removing: %s",
-                         self._stored.unit_path)
+            logger.error(
+                "No systemd unit file found when removing: %s",
+                self._stored.unit_path,
+            )
         systemd.daemon_reload()
         shutil.rmtree(self._stored.agent_path, ignore_errors=True)
 
-    def _check_update_repos_needed(self):
+    def check_update_repos_needed(self):
         """
         Determine if any config settings change which require
         an update to the git repos
         """
-        update_repos = False
-        repo = self.config.get("testflinger-agent-repo")
-        if repo != self._stored.testflinger_agent_repo:
-            self._stored.testflinger_agent_repo = repo
-            update_repos = True
-        branch = self.config.get("testflinger-agent-branch")
-        if branch != self._stored.testflinger_agent_branch:
-            self._stored.testflinger_agent_branch = branch
-            update_repos = True
-        repo = self.config.get("device-agent-repo")
-        if repo != self._stored.device_agent_repo:
-            self._stored.device_agent_repo = repo
-            update_repos = True
-        branch = self.config.get("device-agent-branch")
-        if branch != self._stored.device_agent_branch:
-            self._stored.device_agent_branch = branch
-            update_repos = True
-        if update_repos:
-            self._update_repos()
+        update_needed = False
+        repo = self.config.get("testflinger-repo")
+        if repo != self._stored.testflinger_repo:
+            self._stored.testflinger_repo = repo
+            update_needed = True
+        branch = self.config.get("testflinger-branch")
+        if branch != self._stored.testflinger_branch:
+            self._stored.testflinger_branch = branch
+            update_needed = True
+        if update_needed:
+            self.update_repos()
 
-    def _update_repos(self):
+    def update_repos(self):
         """Recreate the git repos and reinstall everything needed"""
-        tf_agent_dir = f"{self._stored.agent_path}/testflinger-agent"
-        device_agent_dir = f"{self._stored.agent_path}/snappy-device-agents"
-        shutil.rmtree(tf_agent_dir, ignore_errors=True)
-        shutil.rmtree(device_agent_dir, ignore_errors=True)
+        self.cleanup_agent_dirs()
+        repo_path = f"{self._stored.agent_path}/testflinger"
         Repo.clone_from(
-            self._stored.testflinger_agent_repo,
-            tf_agent_dir,
-            multi_options=[f"-b {self._stored.testflinger_agent_branch}"],
+            url=self._stored.testflinger_repo,
+            to_path=repo_path,
+            no_checkout=True,
+            depth=1,
         )
-        self._run_with_logged_errors(
-            [f"{self._stored.venv_path}/bin/pip3", "install", "-I",
-             tf_agent_dir]
+        repo = Repo(repo_path)
```

Comment:
very small optional note, but perhaps you could just do `repo = Repo.clone_from`

Reply:
Done, thanks!